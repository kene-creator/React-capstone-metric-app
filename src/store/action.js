import axios from 'axios';
import { createAsyncThunk } from '@reduxjs/toolkit';

const FETCH_COUNTRY = 'country-name/country/FETCH_COUNTRY';

const API_URL = 'https://restcountries.com/v3.1/region/europe';

const countryList = [];

const countryReducer = (state = countryList, action) => {
  const { type, payload } = action;
  console.log(type, payload);
  switch (type) {
    case 'country-name/country/FETCH_COUNTRY/fulfilled':
      return payload.country;

    default:
      return state;
  }
};

export const fetchCountry = createAsyncThunk(FETCH_COUNTRY, async () => {
  const res = await axios.get(API_URL);
  const resultArray = res.data;
  console.log(resultArray);

  return {
    // eslint-disable-next-line no-unused-vars
    country: resultArray.map((value) => ({
      id: value.area,
      population: value.population,
      name: value.name.official,
      languages: value.languages,
      capital: value.capital,
      image: value.flags.png,
      timezone: value.timezones[0],
      subregion: value.subregion,
    })),
  };
});

export default countryReducer;
