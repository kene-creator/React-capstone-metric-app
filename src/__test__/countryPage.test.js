import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../store/store';
import CountryPage from '../../__mock__/countryPage';

describe('App components snapshots', () => {
  it('Country List renders correctly', () => {
    const tree = render(
      <BrowserRouter>
        <Provider store={store}>
          <CountryPage />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
