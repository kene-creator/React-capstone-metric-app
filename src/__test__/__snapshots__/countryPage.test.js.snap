// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App components snapshots Country List renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex justify-between items-center bg-[#EC4C8A] py-3 px-4"
      >
        <a
          href="/"
        >
          <div>
            <svg
              class="fill-white w-5 h-5 cursor-pointer"
            >
              <use
                xlink:href="icons.svg#icon-chevron-left1"
              />
            </svg>
          </div>
        </a>
        <div
          class="text-white"
        >
          Country
        </div>
        <div
          class="flex "
        >
          <svg
            class="fill-white w-5 h-5 cursor-pointer mr-5"
          >
            <use
              xlink:href="icons.svg#icon-microphone"
            />
          </svg>
          <svg
            class="fill-white w-5 h-5 cursor-pointer"
          >
            <use
              xlink:href="icons.svg#icon-settings"
            />
          </svg>
        </div>
      </div>
      <div>
        <div
          class="flex p-5 bg-[#FC5193] items-center"
        >
          <img
            alt="Country flag"
            class="w-40 h-40 rounded-full ml-8"
          />
          <h3
            class="text-3xl text-white ml-10 font-mono font-bold"
          />
        </div>
        <div
          class="text-white bg-[#DF4782]"
        >
          <p
            class="ml-10"
          >
            Country Information
          </p>
        </div>
        <div>
          <div
            class=" py-10 px-10 bg-[#E94987] text-white flex justify-between items-center"
          >
            <p
              class="text-2xl font-semibold font-mono"
            >
              Capital:
            </p>
            <p />
          </div>
          <div
            class=" py-10 px-10 bg-[#CE4176] text-white flex justify-between items-center"
          >
            <p
              class="text-2xl font-semibold font-mono"
            >
              Population:
            </p>
            <p />
          </div>
          <div
            class=" py-10 px-10 bg-[#E94987] text-white flex justify-between items-center"
          >
            <p
              class="text-2xl font-semibold font-mono"
            >
              Timezone:
            </p>
            <p />
          </div>
          <div
            class=" py-10 px-10 bg-[#CE4176] text-white flex justify-between items-center"
          >
            <p
              class="text-2xl font-semibold font-mono"
            >
              Subregion
            </p>
            <p />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex justify-between items-center bg-[#EC4C8A] py-3 px-4"
    >
      <a
        href="/"
      >
        <div>
          <svg
            class="fill-white w-5 h-5 cursor-pointer"
          >
            <use
              xlink:href="icons.svg#icon-chevron-left1"
            />
          </svg>
        </div>
      </a>
      <div
        class="text-white"
      >
        Country
      </div>
      <div
        class="flex "
      >
        <svg
          class="fill-white w-5 h-5 cursor-pointer mr-5"
        >
          <use
            xlink:href="icons.svg#icon-microphone"
          />
        </svg>
        <svg
          class="fill-white w-5 h-5 cursor-pointer"
        >
          <use
            xlink:href="icons.svg#icon-settings"
          />
        </svg>
      </div>
    </div>
    <div>
      <div
        class="flex p-5 bg-[#FC5193] items-center"
      >
        <img
          alt="Country flag"
          class="w-40 h-40 rounded-full ml-8"
        />
        <h3
          class="text-3xl text-white ml-10 font-mono font-bold"
        />
      </div>
      <div
        class="text-white bg-[#DF4782]"
      >
        <p
          class="ml-10"
        >
          Country Information
        </p>
      </div>
      <div>
        <div
          class=" py-10 px-10 bg-[#E94987] text-white flex justify-between items-center"
        >
          <p
            class="text-2xl font-semibold font-mono"
          >
            Capital:
          </p>
          <p />
        </div>
        <div
          class=" py-10 px-10 bg-[#CE4176] text-white flex justify-between items-center"
        >
          <p
            class="text-2xl font-semibold font-mono"
          >
            Population:
          </p>
          <p />
        </div>
        <div
          class=" py-10 px-10 bg-[#E94987] text-white flex justify-between items-center"
        >
          <p
            class="text-2xl font-semibold font-mono"
          >
            Timezone:
          </p>
          <p />
        </div>
        <div
          class=" py-10 px-10 bg-[#CE4176] text-white flex justify-between items-center"
        >
          <p
            class="text-2xl font-semibold font-mono"
          >
            Subregion
          </p>
          <p />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
